### PR TITLE
Improve collapse and expand of the tree

### PR DESCRIPTION
**Pull request by acidjunk**
The expand of a complete TreeNode and it's children until the next node on the same level works ok.

Todo:
- [x] Fix build
- [x] Add a collapse node
- [x] Clean up some typescript and POC code
- [x] Add some tests

### CODE REVIEW

File: packages/orchestrator-ui-components/src/components/WFOTree/WFOTree.tsx [hunk 1]
```diff
@@ -1,16 +1,27 @@
...
-            <WFOTreeBranch key={item.id} item={item} level={0} />
-        ))}
-    </div>
-);
+export const WFOTree: FC<WFOTreeProps> = ({ data, depthList }) => {
+    const { setDepths } = React.useContext(TreeContext) as TreeContextType;
+
+    useEffect(() => {
+        setDepths(depthList);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (acidjunk) [inline packages/orchestrator-ui-components/src/components/WFOTree/WFOTree.tsx:17]: @ricardovdheijden / @DutchBen : I am not satisfied with this eslint ignore -> changes in the TreeContext will cause a re-render; so it's needed here in the deps.

Or do I miss semething? probably the downside of the Frankenstein state management...

File: packages/orchestrator-ui-components/src/components/WFOTree/styles.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+export const getStyles = () => {
+    const expandIconContainer = {
+        cursor: 'pointer',
+    };
+
+    const treeContainer = {
+        width: 0,
+        marginTop: 8,
+        marginRight: -8,
```
Comment (ricardovdheijden) [inline packages/orchestrator-ui-components/src/components/WFOTree/styles.ts:9]: Try to use the values from the theme for these.

File: packages/orchestrator-ui-components/src/components/WFOSubscription/WFOSubscriptionDetailTree.tsx [hunk 3]
```diff
@@ -22,36 +22,42 @@ import { WFOSubscriptionProductBlock } from './WFOSubscriptionProductBlock';
 
 type NodeMap = { [key: number]: TreeBlock };
 
-const MAX_EXPAND_ALL = 100;
-
 interface WFOSubscriptionDetailTreeProps {
     productBlockInstances: ProductBlockInstance[];
 }
 
+function getNodeDepth(node: TreeBlock, idToNodeMap: NodeMap): number {
```
Comment (ricardovdheijden) [inline packages/orchestrator-ui-components/src/components/WFOSubscription/WFOSubscriptionDetailTree.tsx:29]: could you add some unit tests for this one?

File: packages/orchestrator-ui-components/src/components/WFOTree/WFOTree.tsx (hunk 1 above)
Comment (ricardovdheijden) [reply packages/orchestrator-ui-components/src/components/WFOTree/WFOTree.tsx:17]: You might be able to fix this situation by changing the depthList state (located in the TreeContext) in a useRef.

Comment (acidjunk) [reply packages/orchestrator-ui-components/src/components/WFOTree/WFOTree.tsx]: For now agreed; to try to improve this in upcoming tree tickets: after that decide if we want/need to refactor the Tree context provider.

File: packages/orchestrator-ui-components/src/components/WFOSubscription/WFOSubscriptionDetailTree.tsx (hunk 3 above)
Comment (acidjunk) [reply packages/orchestrator-ui-components/src/components/WFOSubscription/WFOSubscriptionDetailTree.tsx]: yes

File: packages/orchestrator-ui-components/src/components/WfoTree/WfoTreeNode.tsx [hunk 4]
```diff
@@ -38,16 +47,19 @@ export const WfoTreeNode: FC<WfoTreeNodeProps> = ({
     }
 
     return (
-        <div style={{ paddingLeft: `${level * 16}px` }}>
+        <div style={{ paddingLeft: `${level * parseInt(theme.size.m)}px` }}>
```
Comment (ricardovdheijden) [inline packages/orchestrator-ui-components/src/components/WfoTree/WfoTreeNode.tsx:50]: There is also a base value that you do not need to parse

**Review by ricardovdheijden — APPROVED** [review]

**Review by DutchBen — APPROVED** [review]

**Pull request merged**